### PR TITLE
Update encoding.py with some bug fixes

### PR DESCRIPTION
**Pull request by bradfriedman**
Normalize the ordering of the return value of encode_message in encoding.py. Also, some miscellaneous cleaning up.

### CODE REVIEW

File: apitools/base/py/encoding.py [hunk 1]
```diff
@@ -406,7 +406,7 @@ def _DecodeUnknownMessages(message, encoded_message, pair_type):
         new_values.append(new_pair)
     return new_values
 
-
+a
```
Comment (craigcitro) [inline apitools/base/py/encoding.py]: stray character

**Comment by craigcitro** [comment]
Two things:

1. Maybe update the commit description to say what you're fixing -- IIUC, this is just normalizing the order on the return value, right?

2. I'll send a separate email with info on how to make cla-signer happy. :wink:

**Comment by bradfriedman** [comment]
I feel like the lint error here is not useful in this context -- it seems idiomatic to me to check whether a type is in a list of types. Thoughts? Should I disable that rule with an in-line comment?

************* Module apitools.base.py.encoding
C:331,11: Using type() instead of isinstance() for a typecheck. (unidiomatic-typecheck)

**Comment by craigcitro** [comment]
yeah, i agree that the linter warning is spurious here. (we really need the type to be a key in our dict.) you can disable or just bind to a temp var (which confuses the linter :wink:).

i'd say do that and squash, and i'll force cla-signer to wake up if it doesn't chime in.

File: apitools/base/py/encoding.py [hunk 2]
```diff
@@ -328,12 +328,15 @@ def encode_message(self, message):
         if isinstance(message, messages.FieldList):
             return '[%s]' % (', '.join(self.encode_message(x)
                                        for x in message))
-        message_type = type(message)
-        if message_type in _CUSTOM_MESSAGE_CODECS:
+
+        # pylint: disable=unidiomatic-typecheck
```
Comment (craigcitro) [inline apitools/base/py/encoding.py]: small nit: this actually disables the warning for the rest of this block; if you put it at the end of the line, it'll only disable it on that line.

(happily, the linter should be OK with a long line that's due to a pylint disable.)

Comment (bradfriedman) [inline apitools/base/py/encoding.py]: I actually did that at first, then the linter complained because the line was too long :/

**Pull request merged**